src/runtime/builtin_modules/thread.cpp

```diff
@@ -171,16 +171,19 @@ class BoxedThreadLocal : public Box {
         return 0;
     }
 
-    static Box* getattr(Box* obj, char* name) {
-        Box* tls_obj = getThreadLocalObject(obj);
+    static Box* getattrOrNull(Box* obj, char* name) {
+        BoxedDict* tls_obj = static_cast<BoxedDict*>(getThreadLocalObject(obj));
         if (!strcmp(name, "__dict__"))
             return tls_obj;
 
-        try {
-            return getitem(tls_obj, boxString(name));
-        } catch (ExcInfo e) {
+        return tls_obj->getOrNull(boxString(name));
+    }
+
+    static Box* getattr(Box* obj, char* name) {
+        Box* rtn = BoxedThreadLocal::getattrOrNull(obj, name);
+        if (!rtn)
             raiseExcHelper(AttributeError, "'%.50s' object has no attribute '%.400s'", obj->cls->tp_name, name);
-        }
+        return rtn;
     }
 
     static Box* hash(Box* obj) { return boxInt(PyThread_get_thread_ident()); }
```

Comment:
unrelated to this PR, but I just saw this -- should this object really have a different hash on different threads?  looks like CPython doesn't set a custom hash function.